docs/guides/nextjs.md

```diff
@@ -0,0 +1,276 @@
+---
+title: Setup with Next.js
+nav: 21
+---
+
+[Next.js](https://nextjs.org) is a popular server side rendering framework for React that presents
+some unique challenges for using Zustand properly, keep in mind that Zustand store is a global
+variable (AKA module state) making optional the usage of a `Context`, these challenges include:
+
+- **Per-request store:** A Next.js server can handle multiple requests simultaneously. This means
+  that the store should be created per request and should not be shared across requests.
+- **SSR friendly:** Next.js applications are rendered twice, first on the server
+  and again on the client. Having different outputs on both the client and the server will result
+  in a "hydration errors". So the store will have to be initialized on the server and then
+  re-initialized on the client with the same data in order to avoid that. Please read more about
+  that in our [SSR and Hydration](./ssr-and-hygration) guide.
+- **SPA routing friendly:** Next.js supports a hybrid model for client side routing which means
+  that in order to reset a store we need to intialize the store at component level using a
+  `Context`.
+- **Server caching friendly:** Recent versions of Next.js (specifically applications using the App
+  Router architecture) support aggressive server caching. Due to our store is a **module state**
+  is completely compatible with this caching.
+
+We have these general recommendations for appropriate use of Zustand:
+
+- **No global stores** - Because the store is shared across requests, it should not be defined
+  as a global variable. Instead, the store should be created per request.
```

Comment:
@sewera yes, since is a global variable the store is shared across requests and should be created per request

Reply:
So if it should be created per request, maybe let's drop "the store is shared across requests" or change it to say that it should **not** be shared?

```suggestion
- **No global stores** - Because the store should not be shared across requests, it should not be defined
  as a global variable. Instead, the store should be created per request.
```